### PR TITLE
chore: allow discarding beta banner

### PR DESCRIPTION
**Pull request by phanan**
As discussed, this PR adds the ability for the user to discard the "docs current in beta" banner. I've also played a bit with the HTML to make the message fit on devices with smaller screens.

### CODE REVIEW

**Comment by NataliaTepluhina** [comment]
@phanan Netlify build produces an error:

```
ReferenceError: localStorage is not defined
```

**Comment by phanan** [comment]
Oh! I forgot about the client/server part!

**Comment by phanan** [comment]
@NataliaTepluhina Maybe another round?

**Comment by phanan** [comment]
Yeah, I requested reviews from Ben & Sarah as well.

**Comment by phanan** [comment]
@CyberAP Since this is a temporary future, I'd like to keep the component as self-contained as possible.

**Comment by CyberAP** [comment]
> @CyberAP Since this is a temporary feature, I'd like to keep the component as self-contained as possible.

You can achieve the same thing if you put `v-if` on a root element of `<BetaBanner>` component.

**Comment by phanan** [comment]
@CyberAP I think I know I can achieve the same thing with other approaches, thank you :) This is the approach I've chosen though, and I don't think it's worth it to do otherwise.

File: src/.vuepress/components/BetaBanner.vue [hunk 1]
```diff
@@ -1,9 +1,41 @@
...
+      document.querySelector('html').classList.remove('with-beta-banner')
+    }
+  },
+
+  beforeMount() {
+    if (this.shouldShow) {
+      document.querySelector('html').classList.add('with-beta-banner')
+    }
+  }
+}
```
Comment (CyberAP) [inline src/.vuepress/components/BetaBanner.vue:36]:
```suggestion
  data() {
    return {
      shouldShow: typeof window === 'undefined' ?
        true :
        window.localStorage.getItem('beta-banner-discarded') === null
    }
  },

  methods: {
    close() {
      window.localStorage.setItem('beta-banner-discarded', true)
      this.shouldShow = false
      document.querySelector('html').classList.remove('with-beta-banner')
    }
  },

  beforeMount() {
    if (this.shouldShow) {
      document.querySelector('html').classList.add('with-beta-banner')
    }
  }
}
```

**Comment by sdras** [comment]
Thank you An! Looks great!

**Pull request merged**